apps/src/templates/SpeedSlider.jsx

```diff
@@ -0,0 +1,233 @@
+ */
+const SpeedSlider = React.createClass({
+  propTypes: {
+    hasFocus: React.PropTypes.bool,
+    style: React.PropTypes.object,
+    value: React.PropTypes.number.isRequired,
+    onChange: React.PropTypes.func.isRequired
+  },
+
+  getInitialState() {
+    return {
+      dragStart: undefined,
+      valueStart: undefined
+    };
+  },
+
+  componentDidMount() {
+    this.isAndroid_ = dom.isAndroid();
+    this.isIOS_ = dom.isIOS();
+    this.isWindowsTouch_ = dom.isWindowsTouch();
+
+    var thisSlider = this;
+    dom.addMouseDownTouchEvent(this.knob_, function (e) {
+      return thisSlider.onKnobMouseDown(e);
+    });
+    dom.addMouseDownTouchEvent(this.track_, function (e) {
+      return thisSlider.onTrackMouseDown(e);
+    });
+  },
+
+  mouseToSvg_(e) {
+    var svgPoint = this.SVG_.createSVGPoint();
+    // Most browsers provide clientX/Y. iOS only provides pageX/Y.
+    // Android Chrome only provides coordinates within e.changedTouches.
+    if (this.isWindowsTouch_) {
+      // Only screenX/Y properly accounts for zooming in on windows touch.
+      svgPoint.x = e.screenX;
+      svgPoint.y = e.screenY;
+    } else if (this.isAndroid_) {
+      svgPoint.x = e.changedTouches[0].pageX;
+      svgPoint.y = e.changedTouches[0].pageY;
+    } else if (this.isIOS_) {
+      svgPoint.x = e.pageX;
+      svgPoint.y = e.pageY;
+    } else {
+      svgPoint.x = e.clientX;
+      svgPoint.y = e.clientY;
+    }
+    var matrix = this.SVG_.getScreenCTM().inverse();
+    return svgPoint.matrixTransform(matrix);
+  },
+
+  clampValue(val) {
+    return Math.min(Math.max(val, 0), 1);
+  },
+
+  svgPositionToValue(position) {
+    position = (position - knobXMin)/TRACK_LENGTH;
+    return this.clampValue(position);
+  },
+
+  onTrackMouseDown(event) {
+    let mousePosition = this.mouseToSvg_(event);
+    const newValue = this.svgPositionToValue(mousePosition.x);
+    this.props.onChange(newValue);
+    this.setState({
+      dragStart: mousePosition.x,
+      valueStart: newValue
+    });
+    this.startDragging();
+  },
+
+  onKnobMouseDown(event) {
+    let mousePosition = this.mouseToSvg_(event);
+    this.setState({
+      dragStart: mousePosition.x,
+      valueStart: this.props.value
+    });
+    this.startDragging();
+  },
+
+  startDragging() {
+    this.unbindOnMouseMove = dom.addMouseMoveTouchEvent(document, this.onMouseMove);
+    this.unbindStopDragging = dom.addMouseUpTouchEvent(document, this.stopDragging);
+  },
+
+  onMouseMove(event) {
+    let mousePosition = this.mouseToSvg_(event);
+    let mouseDelta = mousePosition.x - this.state.dragStart;
+    let valueDelta = mouseDelta / TRACK_LENGTH;
+    let newValue = this.clampValue(this.state.valueStart + valueDelta);
+    this.props.onChange(newValue);
+  },
+
+  stopDragging(event) {
+    this.setState(this.getInitialState());
+    this.unbindOnMouseMove();
+    this.unbindStopDragging();
+  },
+
+  render() {
+    const props = this.props;
```

Comment:
I don't get any errors, but for some reason I'm having issues using `style` and `hasFocus` within the tags when doing it this way, so I might leave it as is for now.